doc/api/process.md

```diff
@@ -2309,6 +2309,227 @@ import { loadEnvFile } from 'node:process';
 loadEnvFile();
 ```
 
+### Environment variables file parsing specification
+
+This section describes how the environment variables file parsing reads a file
+and sets up the environment variables in Node.js.
+While `.env` files descend from shell scripts that export environment
+variables, there are important distinctions in how they handle quoting,
+spacing, and escaping values.
+Additionally, the [Dotenv][] package  has played a significant role in
+popularizing this format.
+
+#### Basic parsing
+
+* The parser reads line by line, splitting each line into a key and a value
```

Comment:
The parser does not read line by line. It assumes it is a valid line until a newline character is seen.